### PR TITLE
On demand after max attempt

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/EmrResource.scala [hunk 1]
```diff
@@ -242,7 +246,7 @@ object EmrResource {
     .validate[Spec]
     .map { spec =>
       val loggingPath = s"${conf.workflowId}_rsc-${conf.resourceId}_${conf.instanceId}"
-      EmrResource.apply(conf.resourceName, loggingPath, spec)
+      EmrResource.apply(conf.resourceName, loggingPath, spec, conf.instanceId >= conf.maxAttempt)
```
Comment (damianxu88) [inline orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/EmrResource.scala:249]: @anniezhang77 nvm, I can see the `instanceId` is the `instance_attempt`, e.g. `wf-dfd5dc80-957a-428b-91c4-d480dc429133_rsc-c4c36105-cd36-42cc-8038-691fea220731_1` for loggingPath where `1` is the `instanceId`

**Review by damianxu88 — APPROVED** [review]
LGTM

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala [hunk 2]
```diff
@@ -242,7 +242,7 @@ object Ec2Resource {
   def decode(conf: ResourceIO.Conf): JsResult[Ec2Resource] = conf.resourceSpec
     .validate[Spec]
     .map { spec =>
-      Ec2Resource.apply(conf.resourceName, spec)
+      Ec2Resource.apply(conf.resourceName, spec, conf.instanceId >= conf.maxAttempt)
```
[the remark this answers is not in the document]
Comment (damianxu88) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala:245]: @anniezhang77 nvm, saw it in EmrResource now

**Review by schowsf — APPROVED** [review]
LGTM

File: orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (anniezhang77) [reply orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala:245]: yeah it is attempt

**Pull request merged**